dennisreimann/uiengine
# 🍭 ui: add back component rendered source

**Describe the changes**
Fixes #43 
This PR restores the HTML code panel. The rendered HTML is sourced by fetching the document displayed in the preview iframe. It runs at render time of the component instance, and the slight delay between the component setup and content display is handled by a loading state (localized).

A new config option (`componentCommentDelimiter`) has been added to avoid outputting conditional comment blocks around the rendered template: ` ... `. It defaults to true. Setting this flag to false would not output the comment tags in the rendered html; the HTML code panel would show the full HTML document.

This last contribution can be removed if out of scope for this feature, but since I had already coded it I left it as a topic to discuss during the code review.

🛎 Household chores missing: commits should be squashed before merge.

**Steps to check the contribution**
1. Build the project `yarn build`
2. Start the test project `yarn run test-project`
2. Click on Pattern Library/Atoms/Label (http://localhost:3000/patterns/atoms/label/#label-label-html-3)
3. Click on the code label for any variant
4. Click on HTML
5. The rendered HTML code should be displayed after a slight delay

**Checks to pass**
[x] Linting still works
[x] Unit tests are passing
[x] Acceptance tests are passing
[x] Commit messages start with a somehow fitting emoji
